test/kubernetes_unittest.cc

```diff
@@ -137,4 +143,47 @@ TEST_F(KubernetesTest, ComputePodAssociations) {
       ComputePodAssociations(reader, pod->As<json::Object>());
   EXPECT_EQ(expected_associations->ToString(), associations->ToString());
 }
+
+TEST_F(KubernetesTest, GetLegacyResource) {
+  Configuration config(std::stringstream(
+    "KubernetesClusterName: TestClusterName\n"
+    "MetadataApiResourceTypeSeparator: \".\"\n"
+    "InstanceZone: TestZone\n"
+    "InstanceId: TestID\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);  // Don't need HealthChecker.
+  json::value pod = json::object({
+    {"metadata", json::object({
+      {"namespace", json::string("TestNamespace")},
+      {"name", json::string("TestName")},
+      {"uid", json::string("TestUid")},
+    })}
+  });
+  const auto m = GetLegacyResource(reader, json::object({
+    {"metadata", json::object({
+      {"namespace", json::string("TestNamespace")},
+      {"name", json::string("TestName")},
+      {"uid", json::string("TestUid")},
+    })},
+  })->As<json::Object>(), "TestContainerName");
+  EXPECT_EQ(std::vector<std::string>({
+    "gke_container.TestNamespace.TestUid.TestContainerName",
+    "gke_container.TestNamespace.TestName.TestContainerName"
+  }), m.ids());
+  EXPECT_EQ(MonitoredResource("gke_container", {
+    {"cluster_name", "TestClusterName"},
+    {"container_name", "TestContainerName"},
+    {"instance_id", "TestID"},
+    {"namespace_id", "TestNamespace"},
+    {"pod_id", "TestUid"},
+    {"zone", "TestZone"},
+  }), m.resource());
+  EXPECT_TRUE(m.metadata().version.empty());
+  EXPECT_FALSE(m.metadata().is_deleted);
+  EXPECT_EQ(Timestamp(), m.metadata().created_at);
+  EXPECT_EQ(Timestamp(), m.metadata().collected_at);
+  EXPECT_TRUE(m.metadata().ignore);
```

Comment:
Leave this one, remove all of the other checks of `m.metadata()` — they're implementation details.

Reply:
Removed.